### PR TITLE
Cranelift: Get tail calls working on aarch64

### PR DESCRIPTION
**Pull request by fitzgen**

### CODE REVIEW

**Review by jameysharp — COMMENTED** [review]
I haven't reviewed all of this but ran out of time for today. Overall I think it's right but would like somebody else to take a look. For now I have one minor suggestion.

File: cranelift/codegen/src/isa/aarch64/inst/emit.rs [hunk 1]
```diff
@@ -3712,3 +3762,160 @@ impl MachInstEmit for Inst {
...
+        size: OperandSize::Size64,
+        rd: regs::writable_stack_reg(),
+        rn: regs::fp_reg(),
+        imm12: Imm12::maybe_from_u64(off).unwrap(),
+    }
+    .emit(&[], sink, emit_info, state);
+
+    // Move the old FP value from the temporary into the FP register.
+    Inst::AluRRImm12 {
+        alu_op: ALUOp::Sub,
```
[the remark this answers is not in the document]
Comment (fitzgen) [reply cranelift/codegen/src/isa/aarch64/inst/emit.rs:3907]: I originally had it as an add and changed it to a sub to try and get the disassembly to show a `mov`. Neither version worked to get that. *shrug*

[the remark this answers is not in the document]
Comment (afonso360) [reply cranelift/codegen/src/isa/aarch64/inst/emit.rs:3907]: We can use `Inst::gen_move` here right? Hopefully that gives us an actual mov.

**Comment by afonso360** [comment]
Could you also enable tail calls for AArch64 on [fuzzgen](https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/fuzzgen/src/function_generator.rs#L2049)? I've been fuzzing this over that past couple of days and it seems fairly stable.

File: cranelift/codegen/src/isa/aarch64/inst/emit.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (fitzgen) [reply cranelift/codegen/src/isa/aarch64/inst/emit.rs]: D'oh, we can just use `Inst::Move` here.

**Comment by fitzgen** [comment]
> Could you also enable tail calls for AArch64 on fuzzgen, [here](https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/fuzzgen/src/function_generator.rs#L2049)? I've been fuzzing this over that past couple of days and it seems fairly stable.

Good catch, thanks!

Done.

**Review by elliottt — APPROVED** [review]
This looks great to me, thank you for all the comments!

**Comment by afonso360** [comment]
Oops, miss click sorry

**Pull request merged**